### DISCUSSION TITLE
请问一下有关于frp穿透alist的问题

### DISCUSSION THREAD

**Discussion by VeddySama** [AlistGo/alist#1862] [Q&A] [answered]
我用docker-compose搭建的alist，所有配置都是默认配置，用frp去穿透5244端口  但是在服务器上查询不到该端口
PS：frp已穿透过SSH 22 端口  所以frp应该是没问题的

内网机器5244端口正常
![image](https://user-images.githubusercontent.com/49411287/193199159-0b6d5b42-281c-4131-95c6-f1d3992a2ff4.png)

服务器上查询不到5244端口占用
![image](https://user-images.githubusercontent.com/49411287/193199378-7ca9fbcd-d05b-4a7b-8458-86b2a60131e2.png)

内网机器docker-compose配置
![image](https://user-images.githubusercontent.com/49411287/193199455-0fbdef13-e335-4f4c-9bba-8db1a6fe2d7b.png)

内网机器frp配置
![image](https://user-images.githubusercontent.com/49411287/193199539-641c135a-c315-4722-b801-c7919ee439d7.png)

**Comment by rakaki** [comment]
我用樱花frp穿透过没问题

**Reply by VeddySama** [reply]
> 我用樱花frp穿透过没问题

感谢解答  一会我用樱花试一下

**Reply by imRoxy2222** [reply]
请问樱花frp穿透是要选择"建站"吗?要绑定自己域名吗?我这边使用udp和tcp模式穿透都不行

**Comment by VeddySama** [comment, marked as answer]
已经可以通过frp去穿透alist

**Comment by bolavefasfas** [comment]
> I built an alist with docker-compose, all the configurations are default configurations, I use frp to penetrate port 5244 but I can't query the port on the server PS: frp has penetrated SSH port 22, so frp should be no problem
> 
> Intranet machine 5244 port is normal ![image](https://user-images.githubusercontent.com/49411287/193199159-0b6d5b42-281c-4131-95c6-f1d3992a2ff4.png)
> 
> The server can't query the occupied port 5244 ![image](https://user-images.githubusercontent.com/49411287/193199378-7ca9fbcd-d05b-4a7b-8458-86b2a60131e2.png)
> 
> Intranet machine docker-compose configuration ![image](https://user-images.githubusercontent.com/49411287/193199455-0fbdef13-e335-4f4c-9bba-8db1a6fe2d7b.png)
> 
> Intranet machine frp configuration ![image](https://user-images.githubusercontent.com/49411287/193199539-641c135a-c315-4722-b801-c7919ee439d7.png)

can you share the docker-compose.yml full file including the database part for reference

**Reply by VeddySama** [reply]
> > I built an alist with docker-compose, all the configurations are default configurations, I use frp to penetrate port 5244 but I can't query the port on the server PS: frp has penetrated SSH port 22, so frp should be no problem
> > Intranet machine 5244 port is normal ![image](https://user-images.githubusercontent.com/49411287/193199159-0b6d5b42-281c-4131-95c6-f1d3992a2ff4.png)
> > The server can't query the occupied port 5244 ![image](https://user-images.githubusercontent.com/49411287/193199378-7ca9fbcd-d05b-4a7b-8458-86b2a60131e2.png)
> > Intranet machine docker-compose configuration ![image](https://user-images.githubusercontent.com/49411287/193199455-0fbdef13-e335-4f4c-9bba-8db1a6fe2d7b.png)
> > Intranet machine frp configuration ![image](https://user-images.githubusercontent.com/49411287/193199539-641c135a-c315-4722-b801-c7919ee439d7.png)
> 
> can you share the docker-compose.yml full file including the database part for reference

Thanks, the problem has been solved, because of my frp configuration, but I still attach my configuration file

```
alist:
    image: xhofe/alist:v3.0.3
    container_name: alist
    restart: always
    volumes:
      - ./alist/data:/opt/alist/data
      - /file/webdav:/file
    ports:
      - 5244:5244
```
